File: Programming EF Code First/chapter6/Chapter6/Chapter 6 VB/DataAnnotations/DataAccess/CustomConnectionFactory.vb
Imports System.Data.Common
Imports System.Data.Entity.Infrastructure
Imports System.Data.SqlClient

Namespace DataAccess
  Public Class CustomConnectionFactory
	  Implements IDbConnectionFactory
    Public Function CreateConnection(ByVal nameOrConnectionString As String) As DbConnection Implements IDbConnectionFactory.CreateConnection
      Dim name = nameOrConnectionString.Split("."c).Last().Replace("Context", String.Empty)

      Dim builder = New SqlConnectionStringBuilder With {.DataSource = ".\SQLEXPRESS", .InitialCatalog = name, .IntegratedSecurity = True, .MultipleActiveResultSets = True}

      Return New SqlConnection(builder.ToString())
    End Function

  End Class
End Namespace

File: Programming EF Code First/chapter6/Chapter6/BAGA_FluentAPI/DataAccess/CustomConnectionFactory.cs
using System.Data.Common;
using System.Data.Entity.Infrastructure;
using System.Data.SqlClient;
using System.Linq;

namespace DataAccess
{
  public class CustomConnectionFactory : IDbConnectionFactory
  {
    public DbConnection CreateConnection(string nameOrConnectionString)
    {
      var name = nameOrConnectionString
        .Split('.').Last()
        .Replace("Context", string.Empty);

      var builder = new SqlConnectionStringBuilder
      {
        DataSource = @".\SQLEXPRESS",
        InitialCatalog = name,
        IntegratedSecurity = true,
        MultipleActiveResultSets = true
      };

      return new SqlConnection(builder.ToString());
    }
  }
}